### PR TITLE
Patch route testing

### PR DESCRIPTION
**Pull request by shailikadakia**

### CODE REVIEW

File: node-server/__tests__/routesTest.test.js [hunk 1]
```diff
@@ -39,6 +39,77 @@ const testLoc = {
...
+const testLoc2 = {
+    name: 'CMH',
+    description: 'Waterloo Residence Builing',
+    numberOfBathrooms: 0,
+    numberOfBedrooms: 0,
+    address: {
+        streetNumber: 165,
+        streetName: 'University Avenue W',
+        city: 'Waterloo',
+        province: 'Ontario',
```
Comment (anupriyamranjit) [inline node-server/__tests__/routesTest.test.js:51]: Put this into JSON files

File: node-server/__tests__/routesTest.test.js [hunk 2]
```diff
@@ -114,3 +185,96 @@ describe("DELETE Routes", () => {
...
+            },
+            other: {
+                hasGym: true,
+                hasBikeRake: true,
+                hasParking: false,
+                parkingPrice: 0,
+                furnitureIncluded: true,
+                other: 'Hi, welcome to CMH'
+            }
+        }
```
Comment (anupriyamranjit) [inline node-server/__tests__/routesTest.test.js:232]: Put in JSON file

File: node-server/__tests__/routesTest.test.js [hunk 3]
```diff
@@ -114,3 +185,96 @@ describe("DELETE Routes", () => {
...
+                laundryIncluded: true,
+                laundryPrice: 23,
+                internetIncluded: false,
+                internetPrice: null,
+                totalUtilitiesPrice: 68
+            },
+            other: {
+                hasGym: true,
+                hasBikeRake: true,
+                hasParking: false,
```
Comment (anupriyamranjit) [inline node-server/__tests__/routesTest.test.js:269]: Put into JSON file

File: node-server/__tests__/routesTest.test.js [hunk 4]
```diff
@@ -114,3 +185,96 @@ describe("DELETE Routes", () => {
...
+                parkingPrice: 0,
+                furnitureIncluded: true,
+                other: 'Hi, welcome to CMH'
```
Comment (anupriyamranjit) [inline node-server/__tests__/routesTest.test.js:230]: Put into JSON file

File: node-server/api/location.js [hunk 5]
```diff
@@ -131,46 +131,48 @@ router.route('/update/:id').patch(async (req, res) => {
...
+            foundLocation.utilities.internetPrice = internetPrice,
+            foundLocation.utilities.totalUtilitiesPrice = totalUtilitiesPrice,
             foundLocation.other.hasGym = hasGym;
             foundLocation.other.hasBikeRake = hasBikeRake;
             foundLocation.other.hasParking = hasParking;
             foundLocation.other.parkingPrice = parkingPrice;
             foundLocation.other.furnitureIncluded = furnitureIncluded;
             foundLocation.other.other = other;
+            console.log("Before the save");
+            console.log(foundLocation.price, price);
```
Comment (anupriyamranjit) [inline node-server/api/location.js:170]: Remove price console.log

File: node-server/api/location.js [hunk 6]
```diff
@@ -131,46 +131,48 @@ router.route('/update/:id').patch(async (req, res) => {
...
+            foundLocation.utilities.internetIncluded = internetIncluded,
+            foundLocation.utilities.internetPrice = internetPrice,
+            foundLocation.utilities.totalUtilitiesPrice = totalUtilitiesPrice,
             foundLocation.other.hasGym = hasGym;
             foundLocation.other.hasBikeRake = hasBikeRake;
             foundLocation.other.hasParking = hasParking;
             foundLocation.other.parkingPrice = parkingPrice;
             foundLocation.other.furnitureIncluded = furnitureIncluded;
             foundLocation.other.other = other;
+            console.log("Before the save");
```
Comment (anupriyamranjit) [inline node-server/api/location.js:169]: Remove before save

**Review by neels14 — APPROVED** [review]

**Pull request merged**